Description: Fix docshell xpcshell tests to not write to the application directory
Author: Chris Coulson <chris.coulson@canonical.com>
Forwarded: no

Index: firefox-24.0~a2~hg20130707r143486/docshell/test/unit/head_docshell.js
===================================================================
--- firefox-24.0~a2~hg20130707r143486.orig/docshell/test/unit/head_docshell.js	2013-07-08 22:09:48.851753494 +0100
+++ firefox-24.0~a2~hg20130707r143486/docshell/test/unit/head_docshell.js	2013-07-08 22:09:48.843753494 +0100
@@ -6,52 +6,5 @@
 const Ci = Components.interfaces;
 const Cr = Components.results;
 
-var dirSvc = Cc["@mozilla.org/file/directory_service;1"].
-             getService(Ci.nsIProperties);
-var profileDir = dirSvc.get("CurProcD", Ci.nsILocalFile);
-profileDir.append("test_docshell_profile");
-
-// Register our own provider for the profile directory.
-// It will return our special docshell profile directory.
-var provider = {
-  getFile: function(prop, persistent) {
-    persistent.value = true;
-    if (prop == "ProfD") {
-      var retVal = dirSvc.get("CurProcD", Ci.nsILocalFile);
-      retVal.append("test_docshell_profile");
-      if (!retVal.exists())
-        retVal.create(Ci.nsIFile.DIRECTORY_TYPE, 0755);
-      return retVal;
-    }
-    throw Cr.NS_ERROR_FAILURE;
-  },
-  QueryInterface: function(iid) {
-    if (iid.equals(Ci.nsIDirectoryServiceProvider) ||
-        iid.equals(Ci.nsISupports)) {
-      return this;
-    }
-    throw Cr.NS_ERROR_NO_INTERFACE;
-  }
-};
-dirSvc.QueryInterface(Ci.nsIDirectoryService).registerProvider(provider);
-
-function cleanup()
-{
-  // we need to remove the folder that we created for the profile
-  try {
-    if (profileDir.exists())
-      profileDir.remove(true);
-  } catch (e) {
-    // windows has a slight problem with sqlite databases and trying to remove
-    // them to quickly after you might expect to be done with them.  Eat any
-    // errors we'll get.  This should be OK because we cleanup before and after
-    // each test run.
-  }
-}
-
-
-// cleanup from any failed test runs in the past
-cleanup();
-
-// make sure we have our profile directory available to us
-profileDir.create(Ci.nsIFile.DIRECTORY_TYPE, 0755);
+// Needed to properly setup and shutdown the profile.
+do_get_profile();
Index: firefox-24.0~a2~hg20130707r143486/docshell/test/unit/test_nsIDownloadHistory.js
===================================================================
--- firefox-24.0~a2~hg20130707r143486.orig/docshell/test/unit/test_nsIDownloadHistory.js	2013-07-08 06:36:59.000000000 +0100
+++ firefox-24.0~a2~hg20130707r143486/docshell/test/unit/test_nsIDownloadHistory.js	2013-07-08 22:10:53.679754536 +0100
@@ -58,11 +58,6 @@
     return;
   }
 
-  // Needed to properly setup and shutdown the profile.
-  do_get_profile();
-
   for (var i = 0; i < tests.length; i++)
     tests[i]();
-
-  cleanup();
 }
